src/components/views/donate/OnTime/OneTimeDonationCard.tsx

```diff
@@ -311,18 +312,21 @@ const CryptoDonation: FC = () => {
 
 	useEffect(() => {
 		if (
-			amount + gasfee > selectedTokenBalance &&
+			amount > selectedTokenBalance - gasfee &&
+			amount < selectedTokenBalance &&
```

Comment:
i dont think you need this condition `amount  < selectedTokenBalance` because it is going to make the condition to false

Reply:
# Explanation of Transaction Conditions

The removal of the original condition and the addition of the two new conditions are necessary to cover specific scenarios of balance and gas fee more precisely. The condition `amount < selectedTokenBalance` is needed to ensure the transaction amount does not exceed the total balance, while `amount > selectedTokenBalance - gasfee` ensures the transaction is not approved if, after subtracting the gas fee, the remaining balance is not enough to cover the transaction. This prevents the transaction from failing due to insufficient balance to cover the gas fee.

Therefore, both additional conditions are necessary to ensure the transaction is only approved when there is enough balance to cover both the amount and the gas fee, avoiding potential insufficient balance issues. Also, if the transaction exceeds the wallet's balance, a modal will be displayed to inform the user about the insufficient balance. Because this modal will be shown, there is no need to show this error in another way.

### Scenario Explanation

Assume:
- `selectedTokenBalance` = 100 tokens
- `gasfee` = 5 tokens
- `amount` = 95 tokens

- If `amount` were 96 tokens:
  - First condition: 96 tokens > 95 tokens (true)
  - Second condition: 96 tokens < 100 tokens (true)
  - The warning will be shown.

- If `amount` were 105 tokens:
  - First condition: 105 tokens > 95 tokens (true)
  - Second condition: 105 tokens < 100 tokens (false)
  - The modal of insufficient balance will be shown.

### Conclusion

Both conditions ensure:
1. The `amount` does not exceed the total balance.
2. The remaining balance after the gas fee is enough to cover the transaction.

If either condition fails, a modal informs the user about the insufficient balance. Therefore, there's no need to display the error in another way.